lib/withTypescript.js

```diff
@@ -9,7 +9,20 @@ module.exports = (base) => ({
 
     '@typescript-eslint/array-type': ['error', { default: 'generic' }],
     '@typescript-eslint/prefer-optional-chain': ['error'],
-
+    '@typescript-eslint/consistent-type-definitions': ['error', 'interface'],
+    '@typescript-eslint/consistent-type-imports': 'error',
+    '@typescript-eslint/no-floating-promises': 'error',
+    '@typescript-eslint/no-misused-promises': [
+      'error',
+      {
+        'checksVoidReturn': false,
+      },
+    ],
+    '@typescript-eslint/switch-exhaustiveness-check': 'warn',
```

Comment:
Why downgrade to warning only? Adding empty `default` in case it’s intentional imho coveys intention better than keeping a warning (or disabling with a comment).